cdap-common/src/main/java/co/cask/cdap/common/dataset/DatasetClassRewriter.java

```diff
@@ -225,7 +216,7 @@ private Type getMethodAnnotationType(boolean hasRead, boolean hasWrite, boolean
           if (hasWrite) {
             return Type.getType(WriteOnly.class);
           }
-          return Type.getType(NoAccess.class);
+          return null;
```

Comment:
We don't have an unknown annotation class, hence using `null` to represent an un-annotated method.